librz/analysis/dwarf_process.c

```diff
@@ -1398,52 +1315,27 @@ static void apply_debuginfo_variables(FcnVariableCtx *ctx, RzAnalysisVarKind kin
  * \param analysis
  * \param dwarf_sdb
  */
-RZ_API void rz_analysis_dwarf_integrate_functions(RzAnalysis *analysis, RzFlag *flags, Sdb *dwarf_sdb) {
-	rz_return_if_fail(analysis && dwarf_sdb);
-
-	/* get all entries with value == func */
-	SdbList *sdb_list = sdb_foreach_list_filter(dwarf_sdb, filter_sdb_function_names, false);
-	SdbListIter *it;
-	SdbKv *kv;
-	/* iterate all function entries */
-	ls_foreach (sdb_list, it, kv) {
-		char *func_sname = kv->base.key;
-
-		char *addr_key = rz_str_newf("fcn.%s.addr", func_sname);
-		ut64 faddr = sdb_num_get(dwarf_sdb, addr_key, 0);
-		free(addr_key);
-
-		/* if the function is analyzed so we can edit */
-		RzAnalysisFunction *fcn = rz_analysis_get_function_at(analysis, faddr);
-		if (fcn) {
-			rz_analysis_function_delete_arg_vars(fcn);
-			fcn->has_debuginfo = true;
-
-			/* prepend dwarf debug info stuff with dbg. */
-			char *real_name_key = rz_str_newf("fcn.%s.name", func_sname);
-			const char *real_name = sdb_const_get(dwarf_sdb, real_name_key, 0);
-			free(real_name_key);
-
-			char *dwf_name = rz_str_newf("dbg.%s", real_name);
-			rz_analysis_function_rename(fcn, dwf_name);
-			free(dwf_name);
-
-			char *sig_key = rz_str_newf("fcn.%s.sig", func_sname);
-			const char *fcnstr = sdb_const_get(dwarf_sdb, sig_key, 0);
-			free(sig_key);
-			/* Apply signature as a comment at a function address */
-			rz_meta_set_string(analysis, RZ_META_TYPE_COMMENT, faddr, fcnstr);
-		}
-
-		FcnVariableCtx ctx = {
-			.analysis = analysis,
-			.flags = flags,
-			.dwarf_sdb = dwarf_sdb,
-			.func_sname = func_sname,
-			.fcn = fcn,
+RZ_API void rz_analysis_dwarf_integrate_functions(RzAnalysis *analysis, RzFlag *flags) {
+	rz_return_if_fail(analysis && analysis->debug_info);
+	Function *fn;
+	rz_vector_foreach(analysis->debug_info->functions, fn) {
+		/* Apply signature as a comment at a function address */
+		RzType *type = ht_pp_find(analysis->typedb->callables, fn->name, NULL);
```

Comment:
To avoid exposing internal structure of how types are handled, I recommend using the API instead: `RZ_API RZ_BORROW RzCallable *rz_type_func_get(RzTypeDB *typedb, RZ_NONNULL const char *name)`